AutoParkConsole/Classes/Vehicle.cs

```diff
@@ -38,10 +41,24 @@ public Vehicle()
 			TankCapacity = tankCapacity;
 		}
 
-		public double GetCalcTaxPerMonth() => (Weight * 0.0013d) + (VehicleType.TaxCoefficient * 30) + 5;
+		public double GetCalcTaxPerMonth() =>
+			(Weight * 0.0013d) + (VehicleType.TaxCoefficient * Engine.TaxCoefficientByEngineType * 30) + 5;
 
+		public override bool Equals(object obj)
+		{
+			if (obj is Vehicle)
```

Comment:
You can use pattern matching
```suggestion
			if (obj is Vehicle)
```